readme.md

```diff
@@ -39,7 +44,10 @@ There are 4 primitive types in Bru.
 * Boolean
 * Null
 
-A string can contain any character except `:` and `\n`
+The string type slightly differs from JSON in that a Bru string is
+a single-quoted and is always UTF-8. It may contain any printable UTF-8
+character except `'` or `\n`. Whitespace is considered significant inside of the
+quoted string.
```

Comment:
I changed this for multiple reasons:

- examples string values include `:` as part of URLs, so it cannot be true that they cannot contain `:`
- restricting to printable characters makes the behaviour *generally* correct for HTTP headers, query parameters, etc.

I am less comfortable with the quoting rules, but I do not know enough about the Bru language parser to know whether you support escapes (e.g., `'it\'s awesome'`) in strings or whether single quotes are the only supported quotes (`"it's awesome"`). If possible, I would consider copying the [hjson](https://hjson.github.io) rules on strings (some strings can be unquoted).

Reply:
> restricting to printable characters makes the behaviour generally correct for HTTP headers, query parameters, etc.

We would not be able to restrict the contents to just UTF-8 as we also have to support other languages. Ex: see https://github.com/usebruno/bruno/issues/671 in key value pairs

> but I do not know enough about the Bru language parser to know whether you support escapes

Yes, we should support escapes

Reply:
Does JavaScript have good support or non-UTF-8 encoded languages? I know that for a while, the JS string was (is?) UCS-2, which is its own nightmare.

If non-UTF-8 encoding can be supported, I would recommend that some sort of annotation be used. Maybe something like `r'raw string that might be non-utf-8'` as Python does. If not an annotation, perhaps a different delimiter set, like `<<raw string that might not be non-utf-8>>`. Multi-line strings would need a similar distinction `r(\n…\n)` or `<<(\n…\n)>>`.